BaseAdresseNationale/id-fix
# Added webhook mechanism to send error message

# Context

Id-fix is sometimes refusing BAL processing as it does not fulfill the desired format. When detecting errors, it notifies it in logs. In order to have a better view of the error logs, we want to use a hook system to send messages.

# Enhancement

This PR adds a web hook mechanism to send the error messages

# How to test 

- Clone this branch,
- create a mattermost webhook for the test. Mattermost will give you the webhook URL
- add the env variable `MESSAGE_WEBHOOK_URL` in your .env file with the URL previously created
- send a BAL that could trigger an error (for example, a line has a ban_toponyme_id missing). For that, you can use the 'standalone mode' by using the env variable `STANDALONE_MODE=true`
- check that the error message has been sent to the webhook channel
